test/Microsoft.AspNetCore.Mvc.Core.Test/Formatters/StringOutputFormatterTests.cs

```diff
@@ -12,22 +12,31 @@
 
 namespace Microsoft.AspNetCore.Mvc.Formatters
 {
-    public class TextPlainFormatterTests
+    public class StringOutputFormatterTests
     {
-        public static IEnumerable<object[]> OutputFormatterContextValues
+        public static IEnumerable<object[]> CanWriteResultForStringTypesData
         {
             get
             {
-                // object value, bool useDeclaredTypeAsString, bool expectedCanWriteResult
-                yield return new object[] { "valid value", true, true };
-                yield return new object[] { null, true, true };
-                yield return new object[] { null, false, false };
-                yield return new object[] { new object(), false, false };
+                // object value, bool useDeclaredTypeAsString
+                yield return new object[] { "declared and runtime type are same", true };
+                yield return new object[] { "declared and runtime type are different", false };
+                yield return new object[] { null, true };
+            }
+        }
+
+        public static IEnumerable<object[]> CannotWriteResultForNonStringTypesData
+        {
+            get
+            {
+                // object value, bool useDeclaredTypeAsString
+                yield return new object[] { null, false };
+                yield return new object[] { new object(), false };
             }
         }
 
         [Fact]
-        public void CanWriteResult_SetsAcceptContentType()
+        public void CannotWriteResult_ForNonTextPlainOrNonBrowserMediaTypes()
```

Comment:
This name isn't correct. More like `CannotWriteUnsupportedMediaType` since there's only one check here.

Could rename `expectedContentType` to `contentType` (because it's an input) and make this a `[Theory]` that checks a few other unsupported `contentType` values.

Even then, there's no such thing as a "non browser media type". There are browser and non-browser `Accept` headers where the first contain the wildcard media type and the second do not. But, this test doesn't deal with `Accept` headers.